Documentation/project-docs/platform-names.md

```diff
@@ -4,6 +4,26 @@ Over the years, we've accumulated quite a few names, platforms, and components t
+
+* **JIT**. Just in time compiler. This technology compiles IL to machine code that the processor understands. It's called JIT because compilation happens on demand and is performed on the same machine the code needs to run on. Since JIT compilation happens during execution of the application, compile time is part of the run time. Thus, JIT compilers have to trade spending more time optimizing code with the savings the resulting code can produce.
+
+* **AOT**. Ahead of time compiler. Similar to JIT this compiler also translates IL to machine code. However, in contrast to a JIT the compilation happens before the application is executed and is usually performed on a different machine. AOT tool chains don't have to trade run time and compile time and thus can spend more time optimizing. Also, since the context of AOT are entire applications, AOT compiler can also perform cross module linking and whole program analysis.
+
+* **NGEN**. Native (image) generation. You can think of this technology as a persistent JIT compiler. It usually compiles code on the same machine it's supposed to run on, but typically happens during install time.
+
+* **CLR**. Common language runtime. Exact meaning depends on context, but it usually refers to the runtime of the .NET Framework and includes several components. The CLR is a virtual machine, i.e. it includes the facilities to generate and compile code on the fly using a JIT. The existing Microsoft CLR implementation is Windows only.
+
+* **CoreCLR**. Core Common Language Runtime. Built from the same code base as the CLR. Originally, CoreCLR was the runtime of Silverlight and was just always designed to run on multiple platforms, specifically Windows and OS X. CoreCLR is now part of .NET Core and represents a simplified version of the CLR: it's also a virtual machine with a JIT.
+
+* **CoreRT**. Core runtime. In contrast to the CLR/CoreCLR, CoreRT is not a virtual machine, i.e. doesn't include the facilities to generate and run code on the fly because it doesn't include a JIT. It does, however, include the GC and the ability for runtime type identification (RTTI) as well as reflection. However, it's type system is designed so that metadata for reflection can be omitted. This enables having an AOT tool chain that can link away superfluous metadata (and more importantly) identify code that the application doesn't use.
+
+* **CoreFX**. Core framework. Conceptually a set of `System.*` (and to a limited extend `Microsoft.*`) libraries that make up the lower layer of the .NET library stack. It's what most people would think of the BCL (base class library): the most general purpose, lower level set of functionality that higher-level frameworks, such as WCF and ASP.NET, build upon. The source code in the GitHub repo under the same name is the code base of the .NET Core library stack. However, many of the APIs share the same shape with the .NET Framework and are thus accessible when targeting the .NET Framework or .NET Core.
```

Comment:
"It's what most people would think of **as** the BCL"
I don't like "APIs share the same shape" but I can't think of a better way to put it